### PR TITLE
RavenDB-22287 switch TimeoutManager to Task.Delay on low core instances

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22287

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [ ] Low 
- [x] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Sparrow/Utils/TimeoutManager.cs [hunk 1]
```diff
@@ -14,10 +16,20 @@ namespace Sparrow.Utils
 {
     internal static class TimeoutManager
     {
+        private static FrozenDictionary<uint, TimerTaskHolder> ValuesForRead = new Dictionary<uint, TimerTaskHolder>().ToFrozenDictionary();
         private static readonly ConcurrentDictionary<uint, TimerTaskHolder> Values = new ConcurrentDictionary<uint, TimerTaskHolder>();
         private static readonly Task InfiniteTask = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously).Task;
 
-        private sealed class TimerTaskHolder  : IDisposable
+        private const bool ForceTaskDelay = true;
```
Comment (ppekrol) [inline src/Sparrow/Utils/TimeoutManager.cs]: fix me

File: src/Sparrow/Utils/TimeoutManager.cs [hunk 2]
```diff
@@ -14,10 +16,20 @@ namespace Sparrow.Utils
...
+
+        private static readonly bool UseTaskDelay;
+
+        static TimeoutManager()
+        {
+            UseTaskDelay = Environment.ProcessorCount <= 2 || ForceTaskDelay;
```
Comment (ppekrol) [inline src/Sparrow/Utils/TimeoutManager.cs]: maybe env instead?

Comment (ayende) [reply src/Sparrow/Utils/TimeoutManager.cs]: Would probably be better to have that as env yes.

Comment (ayende) [reply src/Sparrow/Utils/TimeoutManager.cs]: We want that only on burstable, basically.

**Review by gregolsky — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**